exporter/datadogexporter/metadata/metadata.go

```diff
@@ -81,25 +92,58 @@ type Meta struct {
 	HostAliases []string `json:"host-aliases,omitempty"`
 }
 
-func getHostMetadata(params component.ExporterCreateParams, cfg *config.Config) *HostMetadata {
-	hostname := *GetHost(params.Logger, cfg)
-	tags := cfg.GetHostTags()
-
-	ec2HostInfo := ec2.GetHostInfo(params.Logger)
-	systemHostInfo := system.GetHostInfo(params.Logger)
-
-	return &HostMetadata{
-		InternalHostname: hostname,
-		Flavor:           params.ApplicationStartInfo.ExeName,
-		Version:          params.ApplicationStartInfo.Version,
-		Tags:             &HostTags{tags},
-		Meta: &Meta{
-			InstanceID:     ec2HostInfo.InstanceID,
-			EC2Hostname:    ec2HostInfo.EC2Hostname,
-			Hostname:       hostname,
-			SocketHostname: systemHostInfo.OS,
-			SocketFqdn:     systemHostInfo.FQDN,
-		},
+// metadataFromAttributes gets metadata info from attributes following
+// OpenTelemetry semantic conventions
+func metadataFromAttributes(attrs pdata.AttributeMap) *HostMetadata {
+	hm := &HostMetadata{Meta: &Meta{}, Tags: &HostTags{}}
+
+	if hostname, ok := HostnameFromAttributes(attrs); ok {
+		hm.InternalHostname = hostname
+		hm.Meta.Hostname = hostname
+	}
+
+	// AWS EC2 resource metadata
+	cloudProvider, ok := attrs.Get(conventions.AttributeCloudProvider)
+	if ok && cloudProvider.StringVal() == conventions.AttributeCloudProviderAWS {
+		ec2HostInfo := ec2.HostInfoFromAttributes(attrs)
+		hm.Meta.InstanceID = ec2HostInfo.InstanceID
+		hm.Meta.EC2Hostname = ec2HostInfo.EC2Hostname
+		hm.Tags.System = ec2HostInfo.EC2Tags
```

Comment:
Just checking, does that mean these tags will be shown as coming from the Datadog Agent, in the infrastructure list (since there's no difference between host tags and EC2 tags)?

Reply:
I changed it to add them to the `OTel` field since I think it makes more sense